File: backend/models/trip.py
from sqlalchemy import Column, Integer, String, Float
from database import Base


class Trip(Base):
    __tablename__ = "trips"

    id                      = Column(Integer, primary_key=True, index=True)
    destination             = Column(String,  nullable=False)
    days                    = Column(Integer, nullable=False)
    budget                  = Column(Float,   nullable=False)
    currency                = Column(String,  nullable=False, default="USD")
    travel_style            = Column(String,  nullable=True)
    travel_month            = Column(String,  nullable=True)
    travel_season           = Column(String,  nullable=True)
    category                = Column(String,  nullable=False)
    recommendation_transport= Column(String,  nullable=False)
    daily_budget            = Column(Float,   nullable=False)

File: frontend/types/trip.ts
// ── Trip domain types ─────────────────────────────────────────────────────────

export type TripCategory = "Backpacker" | "Standard" | "Luxury";

export type TravelStyle = "Backpacker" | "Family" | "Solo" | "Couple" | "Luxury" | "Adventure";

export interface Trip {
  id: number;
  destination: string;
  days: number;
  budget: number;
  currency: string;
  daily_budget: number;
  travel_style: TravelStyle | null;
  travel_month: string | null;
  travel_season: string | null;
  category: TripCategory;
  recommendation_transport: string | null;
  ai_recommendation: string | null;
  created_at?: string;
}

export interface CreateTripPayload {
  destination: string;
  days: number;
  budget: number;
  currency?: string;
  travel_style?: TravelStyle | null;
  travel_month?: string | null;
}

// ── Sort / filter types used by the My Trips dashboard ───────────────────────

export type SortField = "created_at" | "budget";
export type SortOrder = "asc" | "desc";

export interface TripFilters {
  search: string;
  travelStyle: TravelStyle | "";
  sortField: SortField;
  sortOrder: SortOrder;
}
